test-network-function/lifecycle/suite.go

```diff
@@ -104,9 +108,65 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 		}
 
 		testOwner(&configData)
+
+		testScaling(&configData)
 	}
 })
 
+func waitForAllDeploymentsReady(namespace string, timeout, pollingPeriod time.Duration) {
+	gomega.Eventually(func() []string {
+		_, notReadyDeployments := getDeployments(namespace)
+		return notReadyDeployments
+	}, timeout, pollingPeriod).Should(gomega.HaveLen(0))
+}
+
+func testScaling(configData *common.ConfigurationData) {
+	ginkgo.It("Testing deployment scaling", func() {
+		defer results.RecordResult(identifiers.TestScalingIdentifier)
+
+		// Map to register the deployments that have been already tested
+		deploymentNames := make(map[string]bool)
+
+		for _, cut := range configData.ContainersUnderTest {
+			namespace := cut.Oc.GetPodNamespace()
+
+			// Get deployment name and check whether it was already tested.
+			// ToDo: Proper way (helper/handler) to do this.
+			podNameParts := strings.Split(cut.Oc.GetPodName(), "-")
+			deploymentName := podNameParts[0]
+
+			if _, alreadyTested := deploymentNames[deploymentName]; alreadyTested {
+				continue
+			}
+
+			// Get deployment's current replicaCount
+			deployments, _ := getDeployments(namespace)
+			replicaCount := deployments[deploymentName].Replicas
+
+			// ScaleIn, removing one pod from the replicaCount
```

Comment:
should you test if `replicas!=1` before scaling down ?
I know that having ``Replicas  == 1`` would fail other tests.. but not sure if we should push replicas count to 0.

Reply:
I have tested this manually, and it works. After the "oc scale -n tnf deployment test --replicas=0", you get a 0/0 pods deployment:
```
greyerof@polimnia ~/github/test-network-function (CNFCERT-8)$ oc get deployments -n tnf
NAME      READY   UP-TO-DATE   AVAILABLE   AGE
partner   1/1     1            1           81m
test      0/0     0            0           81m
```

I've launched the scaling TC after a manual scale-in to 1 replicas and the TC works ok, so I don't think that check is necessary here.